stub/src/main/java/io/grpc/stub/Calls.java

```diff
@@ -172,10 +172,11 @@
       ReqT param,
       Call.Listener<RespT> responseListener) {
     call.start(responseListener, new Metadata.Headers());
-    call.request(1);
     try {
       call.sendPayload(param);
       call.halfClose();
+      // Request after sending the payload otherwise we prematurely trigger a flush.
```

Comment:
Is this still true after your change to conditionally call flush after returnProcessedBytes?